packages/vectra_detect/data_stream/log/elasticsearch/ingest_pipeline/pipeline-audit.yml

```
@@ -8,38 +8,38 @@ processors:
       field: event.category
       value: session
       allow_duplicates: false
-      if: ctx.json?.message != null && ctx.json.message.contains('session')
+      if: ctx.json?.message?.contains('session') == true
   - append:
       field: event.category
       value: authentication
       allow_duplicates: false
-      if: ctx.json?.message != null && (ctx.json.message.contains('authentication') || ctx.json.message.contains('log in'))
+      if: ctx.json?.message?.contains('authentication') == true || ctx.json?.message?.contains('log in') == true
   - append:
       field: event.category
       value: configuration
       allow_duplicates: false
-      if: ctx.json?.message != null && ctx.json.message.contains('configuration')
+      if: ctx.json?.message?.contains('configuration') == true
   - append:
       field: event.type
       value: end
       allow_duplicates: false
-      if: ctx.json?.message != null && (ctx.json.message.contains('timeout') || ctx.json.message.contains('log in'))
+      if: ctx.json?.message?.contains('timeout') == true || ctx.json?.message?.contains('log in') == true
   - rename:
       field: json.result
       target_field: vectra_detect.log.result
       ignore_missing: true
   - set:
       field: event.outcome
       value: success
-      if: ctx.vectra_detect?.log?.result != null && ['success', 'true'].contains(ctx.vectra_detect.log.result.toLowerCase())
+      if: "['success', 'true'].contains(ctx.vectra_detect?.log?.result?.toLowerCase())"
```

I'm wondering at what point the cross-over is between the ease of an array contains check with lower-casing and the correctness and non-allocation of a duplicated case-insensitive comparison.
```suggestion
      if: 'success'.equalsIgnoreCase(ctx.vectra_detect?.log?.result) || 'true'.equalsIgnoreCase(ctx.vectra_detect?.log?.result)
```